File: test_upload.py
import requests
import json
import base64

def upload(filename):
    with open(filename, 'rb') as f:
        auth = base64.b64encode(b"admin:changeme").decode("utf-8")
        res = requests.post(
            'http://localhost:3000/api/upload',
            files={'files': (filename, f, 'application/vnd.google-earth.kmz')},
            headers={'Authorization': f'Basic {auth}'}
        )
        print(f"Upload {filename}:", res.status_code, res.text)

if __name__ == "__main__":
    upload('test_no_dir.kmz')
    upload('test_dir.kmz')
    res = requests.get('http://localhost:3000/api/layers')
    print("Layers:", res.status_code, res.json())

File: test_upload.js
const fs = require('fs');

async function doUpload() {
  const fileBuffer = fs.readFileSync('test.kmz');
  const blob = new Blob([fileBuffer], { type: 'application/vnd.google-earth.kmz' });
  const formData = new FormData();
  formData.append('files', blob, 'test.kmz');

  const response = await fetch('http://localhost:3000/api/upload', {
    method: 'POST',
    headers: {
      'Authorization': 'Basic ' + btoa('admin:changeme')
    },
    body: formData
  });

  const text = await response.text();
  console.log('Upload response:', text);

  const layersRes = await fetch('http://localhost:3000/api/layers');
  const layers = await layersRes.json();
  console.log('Layers:', layers);
}
doUpload();
